telluric/georaster.py

```diff
@@ -1804,6 +1815,48 @@ def colorize(self, colormap, band_name=None, vmin=None, vmax=None):
 
         return self.copy_with(image=array, band_names=['red', 'green', 'blue'])
 
+    def chunks(self, shape=256, pad=False):
+        """
+        This method returns GeoRaster chunks out of the original raster,
+        The chunck is evaluated only when fetched from the iterator
+
+        Usefult when you want to iterate over a big rasters
+        params:
+
+        shape - int or tuple, the shape of the chunk
+        pad - when set to True all rasters will have the same shape, when False the edge rasters
+              will have a shape less than the requested shape, according to what the raster actually had
+
+        This iterator is over a RasterChucnk namedtuple that has the raster and the offsets in it
+
+        """
+        if isinstance(shape, int):
+            shape = (shape, shape)
+
+        (width, height) = shape
+
+        col_steps = int(self.width/width)
+        row_steps = int(self.height/height)
+        for col_step in range(0, col_steps + 1):
+            col_off = col_step*width
+            if not pad and col_step == col_steps:
+                window_width = self.width % width
```

Comment:
you should check this for a situation where the `width` or `height` of the `raster` is a multiplication of the chunk `width` and `height` respectively

Reply:
why?

Reply:
because I'm afraid that you may loose the last chunk, for example with a raster size (1000, 1000) and shape is (200,200)

Reply:
becasue it might return zero, I get it now

Reply:
I fixed it another way, check it out, I also added to the tests a raster with that multiplication, 

in the end the bug was not losing chunk but creating a chunk with zero width and or height